java/Xpedite/java_setup.cmake

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+################################################################################################
+#
+# Xpedite script to compile Java files to create .jar files and .h files
+#
+# Author: Brooke Elizabeth Cantwell, Morgan Stanley
+#
+################################################################################################
+
+function(java_setup)
+    project(xpedite C CXX ASM)
+
+    find_package(Java REQUIRED)
+    find_package(JNI REQUIRED)
+    include(UseJava)
+    find_package(Java 1.8)
+
+    set(CMAKE_ORIGINAL_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+    set(CMAKE_CURRENT_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/java/Xpedite)
+    get_filename_component(PARENT_DIR ${CMAKE_CURRENT_SOURCE_DIR} DIRECTORY)
+    get_filename_component(PARENT_DIR ${PARENT_DIR} DIRECTORY)
+    set(CMAKE_JAVA_INCLUDE_PATH ${PARENT_DIR}/java/jar/javassist.jar ${JAVA_HOME}/lib/tools.jar)
+    set(JAVA_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src/com/xpedite)
+    file(GLOB JAVA_SOURCE_FILES "${JAVA_SOURCE_DIR}/*.java" "${JAVA_SOURCE_DIR}/probes/*.java")
+    set(XPEDITE_INSTALL_PATH "${PARENT_DIR}/install/java/jar")
+
+    add_jar(Xpedite
+        SOURCES ${JAVA_SOURCE_FILES}
+        OUTPUT_DIR ${XPEDITE_INSTALL_PATH}
+        MANIFEST ${CMAKE_CURRENT_SOURCE_DIR}/src/META-INF/MANIFEST.MF
+    )
+
+    set(CMAKE_INSTALL_PREFIX "java")
+
+    create_javadoc(xpedite-javadocs
+        FILES ${JAVA_SOURCE_FILES}
+        CLASSPATH Xpedite.jar ${CMAKE_JAVA_INCLUDE_PATH}
+        DOCTITLE "Xpedite Java Instrumentation"
+    )
+
+    set(XPEDITE_JNI_HEADER_PATH ${PARENT_DIR}/include/xpedite/java/com_xpedite_Xpedite.h)
+    execute_process(
+        COMMAND javah com.xpedite.Xpedite
+        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/src
+    )
+    file(RENAME ${CMAKE_CURRENT_SOURCE_DIR}/src/com_xpedite_Xpedite.h ${XPEDITE_JNI_HEADER_PATH})
+
```

Comment:
create this under `install/jni` then add `install/jni` to include dir when building `Runtime.C`